src/Data.jl

```diff
@@ -50,25 +53,35 @@ function readfile(filepath::AbstractString, lineparser::Function)::Problem
     return Problem(examples, basename(filepath))
 end
 
+
+"""
+Writes IO examples to disk by serializing them into a file using HDF5 checking for and appending the `.xio` file ending.
+"""
+function write_IOexamples(filepath::AbstractString, examples::Vector{IOExample})
+    serialize(filepath * (endswith(filepath, ".xio") ? "" : ".xio"), examples)
+end
+
+"""
+Writes IO examples and the corresponding programs to disk by serializing them into a file using HDF5 checking for and appending the `.xiop`.
+"""
+function write_IOPexamples(filepath::AbstractString, examples::Vector{Tuple{Data.IOExample, Grammars.Expr}}
```

Comment:
Expr is a built-in Julia type, so you can just use `Expr` instead of `Grammars.Expr`. 
Also, consider the case where the program is just an integer or a variable: In that case, it will not be an expression. 
Therefore, I usually use `Any` for program types. It is not the most elegant solution, but I think any type should be allowed as a program since we can create grammars for arbitrary datatypes.

Reply:
You can then also remove the import for Grammars.